plugins/set.ts

```diff
@@ -25,15 +27,26 @@ function setTag(
 
   // Value is set (e.g. {{ set foo = "bar" }})
   if (expression.includes("=")) {
-    const match = code.match(/^set\s+([\w]+)\s*=\s*([\s\S]+)$/);
+    const match = code.match(/^set\s+([\w{}[\]\s,:.]+)\s*=\s*([\s\S]+)$/);
 
     if (!match) {
       throw new SourceError("Invalid set tag", position);
     }
 
-    const [, variable, value] = match;
+    const variable = match[1].trim();
+    const value = match[2].trim();
     const val = env.compileFilters(tokens, value);
 
+    if (variable.startsWith("{") || variable.startsWith("[")) {
```

Comment:
Probably we should be a bit more careful here; now if someone does `{{ set fo{o = 'bar' }}` we don't throw a particularly nice error.

Reply:
It makes sense. Done.